lib/tasks/migrate_old_primary_pathways.rake

```diff
@@ -0,0 +1,15 @@
+desc 'Takes users in the old primary pathways and moves them to new ones'
+task migrate_old_primary_pathways: :environment do
+  old_developing = Pathway.find_by(slug: 'developing-in-the-classroom')
+  new_developing = Pathway.find_by(slug: 'developing-in-the-classroom-2')
+  old_specialising = Pathway.find_by(slug: 'specialising-or-leading')
+  new_specialising = Pathway.find_by(slug: 'specialising-or-leading-2')
+
+  UserProgrammeEnrolment
+    .where(pathway: old_developing)
+    .update(pathway_id: new_developing.id, message_flags: { primary_pathway_migrated: true })
```

Comment:
Did you mean?
```suggestion
    .update_all(pathway_id: new_developing.id, message_flags: { primary_pathway_migrated: true })
```
`#update` is an undocumented public method. I may be wrong here.

Reply:
update is the same as update on a single record. The potential issue with .update() is that it runs validations and hooks whereas update_all doesn't. update_all() doesn't support getters and setters, IE .update_all(pathway: foo) would fail as "pathway" isn't a column.

I originally wrote it with .update and changed over to passing pathway_id and manually passing the hash so that I could use update_all and it seems like I just forgot to change the method I was calling!